hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -212,21 +192,7 @@ public static List<String> getAllPartitionFoldersThreeLevelsDown(HoodieStorage s
    * Given a base partition and a partition path, return relative path of partition path to the base path.
    */
   public static String getRelativePartitionPath(Path basePath, Path fullPartitionPath) {
-    basePath = CachingPath.getPathWithoutSchemeAndAuthority(basePath);
-    fullPartitionPath = CachingPath.getPathWithoutSchemeAndAuthority(fullPartitionPath);
-
-    String fullPartitionPathStr = fullPartitionPath.toString();
-
-    if (!fullPartitionPathStr.startsWith(basePath.toString())) {
-      throw new IllegalArgumentException("Partition path \"" + fullPartitionPathStr
-          + "\" does not belong to base-path \"" + basePath + "\"");
-    }
-
-    int partitionStartIndex = fullPartitionPathStr.indexOf(basePath.getName(),
-        basePath.getParent() == null ? 0 : basePath.getParent().toString().length());
-    // Partition-Path could be empty for non-partitioned tables
-    return partitionStartIndex + basePath.getName().length() == fullPartitionPathStr.length() ? ""
-        : fullPartitionPathStr.substring(partitionStartIndex + basePath.getName().length() + 1);
+    return getRelativePartitionPath(new StoragePath(basePath.toUri()), new StoragePath(fullPartitionPath.toUri()));
```

Comment:
Let's separate this into a different PR since it's not relevant to fixing the dependency tree?